llpc/util/llpcElfWriter.cpp

```diff
@@ -566,6 +571,8 @@ template <class Elf> void ElfWriter<Elf>::calcSectionHeaderOffset() {
   sharedHdrOffset += m_header.e_phnum * hdrSize;
 
   for (auto &section : m_sections) {
+    if ((strcmp(section.name, RelocName) == 0) && (m_dropRelocSec == true))
```

Comment:
same as above

Reply:
Also, this logic is repeated multiple times in the code. Instead of doing the check in every for loop, can you create a [filter range](https://llvm.org/doxygen/namespacellvm.html#a943f22024cba236f495239754ca3c74b) that does it once and for all? Something like:

```
auto processedSections = 
  make_filter_range(sections, [this] (const Section& s) { return m_dropRelocSec && strcmp(...); });

for (auto &section : processedSections) {
```